ballerina/tests/from_xml_test.bal

```diff
@@ -0,0 +1,628 @@
+}
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapTable2() returns error? {
+    xml x1 = xml `<keys><key>1</key></keys>`;
+    map<table<map<int>>> expected = {"keys": table [{key: 1}]};
+    map<table<map<int>>> actual = check fromXml(x1);
+    test:assertEquals(actual, expected, msg = "testXmlToMapTable2 result incorrect");
+}
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapTable3() returns error? {
+    xml x1 = xml `<keys><key1>1</key1><key2>2</key2></keys>`;
+    map<table<map<int>>> expected = {keys: table [{key1: 1, key2: 2}]};
+    map<table<map<int>>> actual = check fromXml(x1);
+    test:assertEquals(actual, expected, msg = "testXmlToMapTable3 result incorrect");
+}
+
+type Table record {
+    int key1;
+    int key2;
+};
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapTable4() returns error? {
+    xml x1 = xml `<keys><key1>1</key1><key2>2</key2></keys>`;
+    map<table<Table>> expected = {keys: table [{key1: 1, key2: 2}]};
+    map<table<Table>> actual = check fromXml(x1);
+    test:assertEquals(actual, expected, msg = "testXmlToMapTable4 result incorrect");
+}
+
+type BookStores11 record {
+    string ns0\:storeName;
+    int ns0\:postalCode;
+    boolean ns0\:isOpen;
+    Add ns0\:address;
+    Codes11 ns0\:codes;
+    string 'xmlns\:ns0;
+    string status;
+};
+
+type Codes11 record {
+    int[] ns0\:item;
+};
+
+type Add record {
+    string ns0\:street;
+    string ns0\:city;
+    string ns0\:country;
+};
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testToRecord4() returns error? {
+    xml input = xml `<ns0:BookStores11 status="online" xmlns:ns0="http://sample.com/test">
+                        <ns0:storeName>foo</ns0:storeName>
+                        <ns0:postalCode>94</ns0:postalCode>
+                        <ns0:isOpen>true</ns0:isOpen>
+                        <ns0:address>
+                          <ns0:street>foo</ns0:street>
+                          <ns0:city>94</ns0:city>
+                          <ns0:country>true</ns0:country>
+                        </ns0:address>
+                        <ns0:codes>
+                          <ns0:item>4</ns0:item>
+                          <ns0:item>8</ns0:item>
+                          <ns0:item>9</ns0:item>
+                        </ns0:codes>
+                      </ns0:BookStores11>
+                      <!-- some comment -->
+                      <?doc document="book.doc"?>`;
+    BookStores11 expected = {
+        ns0\:storeName: "foo",
+        ns0\:postalCode: 94,
+        ns0\:isOpen:true,
+        ns0\:address: {
+            ns0\:street:"foo",
+            ns0\:city: "94",
+            ns0\:country: "true"
+        },
+        ns0\:codes: {
+            ns0\:item:[4,8,9]
+        },
+        'xmlns\:ns0:"http://sample.com/test",
+        status:"online"
+    };
+    BookStores11 actual = check fromXml(input);
```

Comment:
Same as above. please check

Reply:
Fixed in [821c3fb](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/431/commits/821c3fb1aaf2e91242ee44fc8e5fcdceabc54099)